va/dns_test.go

```diff
@@ -138,7 +138,7 @@ func TestDNSValidationNoServer(t *testing.T) {
 	va, log := setup(nil, 0, "", nil)
 	va.dnsClient = bdns.NewTest(
 		time.Second*5,
-		nil,
+		bdns.NewStaticProvider([]string{}),
```

Comment:
I have a slight preference for `nil` here.

Reply:
Currently, [//bdns/dns.go:282](https://github.com/letsencrypt/boulder/pull/5360/files#diff-4d70665b69d43378fd22771ee9c15d3a7c5fa14dcb797aed69036ac2eb6bcb72R282) unconditionally calls `dnsClient.servers.Addr()`, which results in a nil pointer exception if this static provider isn't configured. I could add code there to make it check that the `.servers` field is non-nil, but simply populating it with a provider that always returns an empty set of addresses felt more elegant.